File: JavaScript solutions/204. Count Primes.js
/**
 * @param {number} n
 * @return {number}
 */
var countPrimes = function (n) {
  var visit = new Array(n + 1).fill(false);

  for (var i = 2; i * i <= n; i++) {
    if (!visit[i]) {
      for (var j = i * i; j <= n; j += i) {
        visit[j] = true;
      }
    }
  }

  var cnt = 0;
  for (var i = 2; i < n; i++) {
    if (!visit[i]) {
      cnt++;
    }
  }

  return cnt;
};

File: C++ solutions/204. Count Primes.cpp
class Solution {
public: 
    vector<bool>visit;
    int countPrimes(int n) {
        visit.resize(n + 1, false);
        for(int i=2;i*i<=n;i++){
            if(!visit[i]){
                for (int j = i * i; j <= n; j += i) {
                    visit[j] = true;
                }
            }
        }
        int cnt=0;
        for(int i=2;i<n;i++) if(!visit[i]) cnt++;
        return cnt;;
    }
};